### DISCUSSION TITLE
No access to Email configuration

### DISCUSSION THREAD

**Discussion by dominikjasek** [strapi/strapi#6458] [Legacy Topics]
**Describe the bug**
Hey, so I just wanted to set email configuration to be able to send email notification using SMTP with Gmail. I followed this tutorial:

https://strapi.io/documentation/3.0.0-beta.x/plugins/email.html#configure-the-plugin

I can´t acces my developer tab:
http://localhost:1337/admin/plugins/email/configurations/development

**Screenshots**
![image](https://user-images.githubusercontent.com/48070343/83332958-70400e00-a29e-11ea-9b98-db4a105dce30.png)

The react error link is refering here:
https://reactjs.org/docs/error-decoder.html/?invariant=130&args[]=null&args[]=

Is there any better solution to send email notification from serer or through SMTP than using strapi? 

Help me, please, thanks :)

**Comment by dominikjasek** [comment]
Alright, I can see I am seeing deprecated version...

**Comment by nourabbas96** [comment]
@holyduke did you solve it

**Comment by dominikjasek** [comment]
Well not yet. I struggle to authenticate sendgrid...

They state [here](https://strapi.io/documentation/v3.x/plugins/email.html#configure-the-plugin) that I should insert that code into ./config/plugins.js

That file does not exist tho. When I created one it didn´t work.

**Comment by stewy-888** [comment]
@holyduke just going to echo that I am similarly having this issue - the docs seem to be outdated since it's pointing out a cog or call to action to bring up the email provider config settings but it's not there. I am also running into the issue that you're running into with the react pages blowing up when directly navigating to http://localhost:1337/admin/plugins/email/configurations/development

![image](https://user-images.githubusercontent.com/12414656/83337315-14aa5c00-a26f-11ea-9f8c-6d824b985970.png)

Tried in both Chrome/Firefox latest
node version: v14.2.0
npm version: 6.14.4
package.json

`{
    "knex": "<0.20.0",
    "pg": "latest",
    "strapi": "3.0.0",
    "strapi-admin": "3.0.0",
    "strapi-connector-bookshelf": "3.0.0",
    "strapi-plugin-content-manager": "3.0.0",
    "strapi-plugin-content-type-builder": "3.0.0",
    "strapi-plugin-documentation": "3.0.0",
    "strapi-plugin-email": "3.0.0",
    "strapi-plugin-upload": "3.0.0",
    "strapi-plugin-users-permissions": "3.0.0",
    "strapi-provider-email-sendgrid": "^3.0.1",
    "strapi-utils": "3.0.0"
  }`

**Comment by dominikjasek** [comment]
@chamchu Sad to hear that, hopefully someone might help us...

**Comment by dominikjasek** [comment]
So I figured it out using vannila sendgrid, here´s what you need to do:

1. helpful tutorial, but it requires additional changes https://www.youtube.com/watch?v=s2bzUzHeSVw
2. `npm i @sendgrid/mail` and  `npm install dotenv --save`
4. make .env file and put SENDGRID_API_KEY =YOUR_API_KEY_FROM_SENDGRID_INSERT_HERE
6. this will send an email:
```
    const sgMail = require("@sendgrid/mail");
    require('dotenv').config()
    try {
        sgMail.setApiKey(process.env.SENDGRID_API_KEY);
        const msg = {
            to: "xxx@yyy.zzz",
            from: "xxx@yyy.zzz",
            subject: "test",
            text: "this is a text",
            html: "<strong>stron text</strong>",
        }
        sgMail.send(msg).then(() => {
            console.log('Message sent')
        }).catch((error) => {
            console.log(error.response.body)
        })
      console.log("email sent");
    } catch (err) {
      console.log(err.message);
      return err;
    }
```

hope you´ll manage to make it work

**Comment by derrickmehaffy** [comment]
Hello,

I see you are wanting to ask a question that is not really a bug report or a feature request, questions should be directed to the [discussions](https://github.com/strapi/strapi/discussions). Please see the following contributing guidelines for asking a question [here](https://github.com/strapi/strapi/blob/master/CONTRIBUTING.md#reporting-an-issue).

I have converted this to a discussion for you, we try to keep Github clean and only for bug reports and feature requests.

**Reply by stewy-888** [reply]
Thanks @derrickmehaffy - this straddles the bug/question fence. What would you suggest then if a feature that's missing from a basic config standpoint like it should be when configuring the email settings out of the box? (eg, https://stackoverflow.com/questions/56112400/send-emails-using-strapi)

The steps I took are: pull down strapi, create the basic back end via commands suggested via the tutorial. Next step was going into trying to configure the email right away but no such cog exists.

**Reply by derrickmehaffy** [reply]
@chamchu I didn't read through the whole issue, but I saw that you weren't using one of the required templates.

Thus I moved it over to a discussion. This page: https://github.com/strapi/strapi/issues/new/choose clearly defines what each option is.

A bug report is something that is causing a failure, in which case the required template must be filled out (at least the steps to reproduce, and the system information). As for a feature request, that template is also required to ensure nothing like it already exists on the product board, and likewise to hopefully encourage people to post the FR there (as well as github).

If it doesn't fit either of the two above, it goes to discussions as a question.

**Reply by stewy-888** [reply]
Got it, thanks!

**Comment by stewy-888** [comment]
@holyduke @nourabbas96 btw, after poking around: https://github.com/strapi/strapi/pull/6195

Looks like it was intended.,

**Reply by stewy-888** [reply]
I guess my gripe is both reason and alternative seems to be undocumented. Spent an hour on this.

**Comment by mountiny** [comment]
@chamchu @holyduke @derrickmehaffy 
Hi, I have been stuck with this problem for couple of hours now. I want to use the classic functionality as resetpassword, sendconfirmation email, so it does not make sense for me to rewrite all of it with vanilla sendgrid. 

Do you have any idea, when this should be resolved? Basically, I cant access the email configuration now to switch the providers so I keep getting 400 Bad request.

**Reply by stewy-888** [reply]
hi @mountiny -- can you tell me the location of your plugins.js file that contains your email config? 

![image](https://user-images.githubusercontent.com/12414656/84294536-29ec7980-aafe-11ea-9751-667f1c2bfff9.png)

Another thing you can try is this-- go into your packages folder and find `strapi-provider-email-sendgrid` update index.js and log the error to see what is going on in the handler.

**Reply by mountiny** [reply]
hi @chamchu - I have just couple minutes ago found out part of the problem. I have not solved it yet totally. My problem was that as I followed the tutorial in documentation the npm install of strapi-email-provider-sendgrid version was not the latest (it was 3.0.0beta....) whereas as I create the strapi backend with npx couple days ago, all of my other packages had 3.0.1. which I did not notice at the beginning. 

After changing the sendgrid provider version to 3.0.1 it no longer throws error on localhost.

However, weirdly enough, it says the email has been send fine, but I do not receive any emails and in the Sendgrid dashboard it says 0 requests have been handled. I have correct api key in the config, do you have any idea, what could be wrong? Thank you very much!

**Reply by stewy-888** [reply]
You will need to add and verify your sender email (eg. your "from" address and "replyto" address in sendgrid, otherwise sendgrid will block you from using their API to send anything)

**Reply by stewy-888** [reply]
@mountiny if you log the error in the `index.js` of the sendgrid provider (in packages) you will find a potential error like this:
`The from address does not match a verified Sender Identity. Mail cannot be sent until this error is resolved. Visit https://sendgrid.com/docs/for-developers/sending-email/sender-identity/ to see the Sender Identity requirements`

**Reply by mountiny** [reply]
@chamchu Thank you for help. I have tried to add the sender as per official guides https://sendgrid.com/docs/ui/sending-email/sender-verification/#adding-a-sender However, in my sendgrid app settings there is not Single Sender Verification showing up:
![image](https://user-images.githubusercontent.com/36083550/84297620-bf5f2c00-ab45-11ea-9535-a23cb176c8d8.png)

**Reply by stewy-888** [reply]
That doesn't look normal to me - you may have to contact sendgrid support. I haven't seen that before :(

**Reply by stewy-888** [reply]
![image](https://user-images.githubusercontent.com/12414656/84298160-8605cc80-ab03-11ea-93c1-8a392712f2d1.png)
what my dashboard looks like

**Reply by mountiny** [reply]
@chamchu Thank you for your help! I have ended up removing the sendgrid addon and trying to install it back again, but it told me this user is banned. So I have contacted the heroku support.

**Comment by mountiny** [comment]
@derrickmehaffy I believe, this should be considered as a bug, because if you follow the current documentation for enabling sendgrid email provider and use that one instead the default one it just does not work and leads to a failure. You cannot even do some of the steps described in documentation. 

To reproduce this bug - start new strapi project and follow documentation to add sendgrid provider.

Thank you for any help
